devstack/devstack.dockerfile

```diff
@@ -0,0 +1,68 @@
+FROM ubuntu:xenial
+ARG USER_NAME
+ENV USER_NAME ${USER_NAME:-root}
+ARG USER_ID
+ENV USER_ID ${USER_ID:-0}
+
+#
+# Install dependencies (this should all be in debian_packages.lst but not sure yet why it's not already)
+#
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y git sudo python python3-dev python3 python3-venv ca-certificates unzip openssh-server mongodb mysql-server mysql-client unzip gcc make
+#
+# Install firefox 52 because it is selenium < 3 compatible
+#
+RUN apt-get install -y software-properties-common && add-apt-repository --yes ppa:jonathonf/firefox-esr-52 && apt-get update && apt-get install -y firefox-esr
+#
+# Install postgresql 10 because circle.yml has that version mentionned, not sure it would make a difference to keep 9
+# 
+RUN echo 'deb http://apt.postgresql.org/pub/repos/apt/ xenial-pgdg main' >> /etc/apt/sources.list.d/pgdg.list && wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y postgresql-10
+#
+# All commands are run via exec under a non privileged user that has the same name as the user building the image
+# and the same id. So that volumes mounted from a directory they own are not unexpectedly chowned to root in various
+# places.
+#
+RUN if test $USER_NAME != root ; then useradd --no-create-home --home-dir /tmp --uid $USER_ID $USER_NAME && echo "$USER_NAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers ; fi
+#
+# en_US.UTF-8 is assumed to always be installed
+#
+RUN apt-get install -y locales && locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8
+#
+# Install consul (should be in a makefile target common to circle.yml and devstack)
+#
+RUN wget -P /tmp https://releases.hashicorp.com/consul/1.2.1/consul_1.2.1_linux_amd64.zip && sudo unzip /tmp/consul_1.2.1_linux_amd64.zip -d /usr/local/bin
+COPY devstack/consul.service /etc/systemd/system/consul.service
+RUN systemctl enable consul
+#
+# Install system dependencies (should be the only place where an install happens)
+#
+WORKDIR /opt
+COPY Makefile .
+COPY debian_packages.lst .
+RUN mkdir devstack && touch devstack/Makefile && make install_system_dependencies
+#
+# Setup a virtual environment owned by the user with all python dependencies installed
+#
+COPY requirements.txt .
+COPY cleanup_utils/requirements.txt cleanup_utils/requirements.txt
+RUN python3 -m venv venv && . venv/bin/activate && pip install --upgrade pip && pip install --upgrade virtualenv && pip install -r requirements.txt && pip install -r cleanup_utils/requirements.txt && chown -R $USER_NAME venv
```

Comment:
Since we're on a isolated container running a single application, we don't need a virtualenv, specially when building container directly from `python` builds.

Reply:
IIRC there are (very) subtle differences when running outside of a virtualenv. This is only a vague recollection. If I'm not mistaken Ocim runs from a virtualenv in production and that circle.yml also installs a virtualenv for testing purposes. If that's not a blocker, I'd rather not try to verify if running outside of a virtualenv makes a difference in the context of this pull request. Maybe it's just me being over-cautious and not trying something that nobody does ;-)